graphql.go

```diff
@@ -359,16 +355,14 @@ func (c *Client) executeRequest(gr *graphResponse, r *http.Request) error {
 
 		if len(gr.Errors) > 0 {
 			// Check to see if we should retry based on error field
-			if shouldRetry(gr.Errors) {
+			if retryConfig.Interval <= c.retryConfig.Interval && shouldRetry(gr.Errors) {
```

Comment:
Something I noticed is that if retryConfig policy were to be set to `Linear` instead of `ExponentialBackoff`, `retryConfig.Interval` will not be increased inside `c.sendRequest(&retryConfig, r)`, line 343. Therefore retries could take place there as well as here. I didn't know if that is a desirable beharior and thought to point it out.

Reply:
Duy pointed out that there's a separate issue with `c.retryConfig.Interval` that he will fix later. This change at least seems to fix the issue with requests silently failing after max attempts